### PR TITLE
rawspeed: Leica DIGILUX 3 support

### PR DESCRIPTION
**Pull request by kmilos**
Addresses https://github.com/darktable-org/darktable/issues/10088

### CODE REVIEW

**Comment by kmilos** [comment]
@rhurlin You can easily test this by patching your cameras.xml and let us know if it works for you; no changes to source code are needed luckily since adobe_coeffs.c is already populated for this camera...

File: data/cameras.xml [hunk 1]
```diff
@@ -5868,6 +5868,17 @@
...
 	</Camera>
+	<!-- Leica Digilux 3 is the same camera as L1 -->
+	<Camera make="LEICA" model="DIGILUX 3" decoder_version="2">
+		<ID make="Leica" model="Digilux 3">Leica DIGILUX 3</ID>
+		<Crop x="0" y="0" width="-11" height="-1"/>
+		<Sensor black="0" white="3986"/>
+	</Camera>
+	<Camera make="LEICA" model="DIGILUX 3" mode="4:3" decoder_version="2">
+		<ID make="Panasonic" model="Digilux 3">Leica DIGILUX 3</ID>
+		<Crop x="0" y="0" width="-11" height="-1"/>
```
Comment (LebedevRI) [inline data/cameras.xml:5879]: Crop should always be a rounded up to the CFA size + 1
I.e. if CFA is 2x2, and crop is +1,+1, it should actually be +4,+4.

Comment (kmilos) [reply data/cameras.xml:5879]: Should or must? So it was "wrong" for the Panasonic DMC-L1 as well? Fix both (potential break of existing L1 edits?), or diverge?

Comment (kmilos) [reply data/cameras.xml:5879]: And btw, the sensor has an odd 3177x2367 width and height, so the odd crop actually makes sense here.

**Comment by LebedevRI** [comment]
@kmilos have you checked that the image actually loads properly afterwards?

**Comment by LebedevRI** [comment]
> @kmilos have you checked that the image actually loads properly afterwards?

Never mind, i see the problem, i forgot to check that #307 doesn't break older RW2's...

File: data/cameras.xml [hunk 2]
```diff
@@ -5868,6 +5868,17 @@
...
 		<Sensor black="0" white="3986"/>
 	</Camera>
+	<!-- Leica Digilux 3 is the same camera as L1 -->
+	<Camera make="LEICA" model="DIGILUX 3" decoder_version="2">
+		<ID make="Leica" model="Digilux 3">Leica DIGILUX 3</ID>
+		<Crop x="0" y="0" width="-11" height="-1"/>
+		<Sensor black="0" white="3986"/>
+	</Camera>
+	<Camera make="LEICA" model="DIGILUX 3" mode="4:3" decoder_version="2">
+		<ID make="Panasonic" model="Digilux 3">Leica DIGILUX 3</ID>
```
Comment (LebedevRI) [inline data/cameras.xml:5878]: So is it Leica or Panasonic?

**Comment by LebedevRI** [comment]
@kmilos thank you.

**Comment by kmilos** [comment]
Thanks, and sorry for the copy/paste induced faux pas 🤦

**Pull request merged**